py4j-python/src/py4j/java_gateway.py

```diff
@@ -794,7 +936,32 @@ def __call__(self, *args):
         return return_value
 
 
-class JavaPackage():
+class UserHelpAutoCompletion(object):
```

Comment:
I am very much open to suggestions on the best way to deal with this. 

The use case I am working towards is users who don't really know what to type where, they certainly won't have read Py4J manual. So in places where it is not possible to provide completions, I wanted to give a clue to the user as to what they could type next. Unfortunately I can't say that this is a pattern or common use case in IPython or PyDev.